pkg/vault/vault.go

```diff
@@ -0,0 +1,209 @@
+
+	s := &Service{
+		vault:        client,
+		appName:      appName,
+		deploymentID: deploymentID,
+	}
+
+	ok, err := s.transitKeyExists()
+	if err != nil {
+		panic(err)
+	}
+
+	if !ok {
+		err = s.transitKeyCreate()
+		if err != nil {
+			panic(err)
+		}
+	}
+	return s
+}
+
+func (vs *Service) keyPath(scope string) string {
+	return fmt.Sprintf("kv/%s/%s/%s", vs.deploymentID, vs.appName, scope)
+}
+
+func (vs *Service) transitKeyName() string {
+	return fmt.Sprintf("%s_kaigara_%s", vs.deploymentID, vs.appName)
+}
+
+func (vs *Service) transitKeyExists() (bool, error) {
+	secret, err := vs.vault.Logical().Read("transit/keys/" + vs.transitKeyName())
+	if err != nil {
+		return false, err
+	}
+	return secret != nil, nil
+}
+
+func (vs *Service) transitKeyCreate() error {
+	_, err := vs.vault.Logical().Write("transit/keys/"+vs.transitKeyName(), map[string]interface{}{
+		"force": true,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Encrypt the plaintext argument and return a ciphertext string or an error
+func (vs *Service) Encrypt(plaintext string) (string, error) {
+	secret, err := vs.vault.Logical().Write("transit/encrypt/"+vs.transitKeyName(), map[string]interface{}{
+		"plaintext": base64.URLEncoding.EncodeToString([]byte(plaintext)),
+	})
+	if err != nil {
+		return "", err
+	}
+
+	ciphertext, ok := secret.Data["ciphertext"]
+	if !ok {
+		return "", fmt.Errorf("ciphertext not found in vault types.Response")
+	}
+	return ciphertext.(string), nil
+}
+
+// Decrypt the given ciphertext and return the plaintext or an error
+func (vs *Service) Decrypt(ciphertext string) (string, error) {
+	secret, err := vs.vault.Logical().Write("transit/decrypt/"+vs.transitKeyName(), map[string]interface{}{
+		"ciphertext": ciphertext,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	data, ok := secret.Data["plaintext"]
+	if !ok {
+		return "", fmt.Errorf("plaintext not found in vault types.Response")
+	}
+
+	plaintext, err := base64.URLEncoding.DecodeString(data.(string))
+	return string(plaintext), err
+}
+
+// LoadSecrets loads existing secrets from vault
+func (vs *Service) LoadSecrets(scope string) error {
+	fmt.Println("Loading secrets...")
+	secret, err := vs.vault.Logical().Read(vs.keyPath(scope))
+	if err != nil {
+		return err
+	}
+
+	// fmt.Println("Secret:", secret)
+
+	if vs.data == nil {
+		vs.data = make(map[string]interface{})
+	}
+
+	if secret == nil || secret.Data == nil {
+		// fmt.Println("Secret or data is nil, overwriting:", vs.data[scope])
+		vs.data[scope] = make(map[string]interface{})
+	} else {
+		// fmt.Println("Secret is not nil, writing", secret.Data["data"])
+		vs.data[scope] = secret.Data["data"].(map[string]interface{})
+	}
+
+	return nil
+}
+
+// SetSecret stores all secrets into the memory
+func (vs *Service) SetSecret(name string, value interface{}, scope string) error {
+	// Secret scope only supports strings
+	if scope == "secret" {
+		encrypted, err := vs.Encrypt(fmt.Sprintf("%v", value))
```

Comment:
better to check that this is actually a string:
`
str, ok := values.(string)
if !ok {
return fmt.Errors("")
}
`